backend/src/packages/files/file.repository.ts

```diff
@@ -0,0 +1,88 @@
+import { type Repository } from '#libs/types/types.js';
+import { FileEntity } from '#packages/files/file.entity.js';
+import { type FileModel } from '#packages/files/file.model.js';
+
+import {
+  type FileCommonQueryResponse,
+  type FileCreateQueryPayload,
+} from './libs/types/types.js';
+
+class FileRepository implements Repository {
+  private fileModel: typeof FileModel;
+
+  public constructor(fileModel: typeof FileModel) {
+    this.fileModel = fileModel;
+  }
+
+  public find(): ReturnType<Repository['find']> {
+    return Promise.resolve(null);
+  }
+
+  public async findById(id: number): Promise<FileEntity | null> {
+    const file = await this.fileModel
+      .query()
+      .findById(id)
+      .castTo<FileCommonQueryResponse | undefined>()
+      .execute();
+
+    if (!file) {
+      return null;
+    }
+
+    return FileEntity.initialize({
+      id: file.id,
+      url: file.url,
+      contentType: file.contentType,
+      createdAt: new Date(file.createdAt),
+      updatedAt: new Date(file.updatedAt),
+    });
+  }
+
+  public async findAll(): Promise<FileEntity[]> {
```

Comment:
Do we need this method? I think it's unnecessary to implement it. You can return Promise.resolve() for this.

Reply:
since files is a separate table - we might need it in the future. I won't delete It for now, but I agree that the code is redundant a bit...

Reply:
So, I think we need coaches' thoughts about it, because I implemented the surveys endpoint and got the same comment from @Liza-Veis here [https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/pull/66#discussion_r1306434148](url)

Reply:
yep, we don't need to implement methods we don't use

Reply:
but let's leave findById, as we will use it in next step

Reply:
But we don't need find all, so it's better to remove it

Reply:
Yes, is it about both findById and findAll?
Because, it seems to me, that findAll won't be used in future.

Reply:
Yes, only findAll